### DISCUSSION TITLE
Search in sections only

### DISCUSSION THREAD

**Discussion by Darrelk** [squidfunk/mkdocs-material#3311] [General]
Hello,

I would like to know if there is a way to configure mkdocs-material to search within a section only, and not across the entire site?

Sample navigation config:

```
nav:
  - Home: index.md
  - User guides:
    - Getting started: UserGuides/GettingStarted.md 
    - Installation:
      - Install windows: UserGuides/installWin.md
      - Install mac: UserGuides/installMac.md    
  - Integration:
      - Doc 1: Integration/Doc1.md 
      - Doc 2: Integration/Doc2.md 
      - Doc 3: Integration/Doc3.md 
      - Doc 4: Integration/Doc4.md 
```

Using the above configuration, I would like to search all sections when on the home page, but only the main section if in child documents. So all pages within the "User guides" section when on any of those pages, and no results in the "integration" section if not in that section. If this makes sense?

**Comment by squidfunk** [comment]
Some options:

- You could build different MkDocs sites per section and combine them into one, so all search indexes are separate.
- You could patch the [query transform](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/#query-transformation) and always add a term that only occurs in a section, similar to `query +section`. The `+` modifier ensures that the term must occur. You could also use [tags](https://squidfunk.github.io/mkdocs-material/setup/setting-up-tags/) to identify sections.

**Reply by dobl1** [reply]
Hi @squidfunk,

Interested in the same behaviour.
It is possible to search filtering by Tag ? like `tag:something search terms` or even adding a list of tags to the search engine with buttons / list (filters) to search only on pages thats contains this tag(s).

Thank you for your work and this incredible tool (I talk about it a lot around me, and push into my compagny (OSS software publisher) to be a sponsor)

**Reply by squidfunk** [reply]
Yes, tags are integrated with search! If you tag your pages, they will be discoverable by tag:
https://squidfunk.github.io/mkdocs-material/setup/setting-up-tags/#__tabbed_2_2

If you add a `+` before the tag name when searching, only pages with that tag will be included.

**Reply by rmscode** [reply]
Apologies for the necro post. This happened to come up as a top search result . . . 

Is this not a thing anymore (`+` before a tag in search)? I just enabled tags and added a few, but I can't seem to search by `#tag` only.

**Reply by squidfunk** [reply]
Still works:

<img width="705" alt="Bildschirm­foto 2025-03-28 um 13 16 56" src="https://github.com/user-attachments/assets/f2c2e047-eb92-428e-ac54-16259aadb547" />

(I've just added the following to a random page to test it):

``` yaml
---
tags: [foo, bar]
---
```

`+{tag}` works a well, but you it might also return results for where the tag is used in body copy or titles. Other than that, if your tag names are unique, it will only return that tag:

<img width="708" alt="Bildschirm­foto 2025-03-28 um 13 20 27" src="https://github.com/user-attachments/assets/866e6375-f2d0-42bb-a5a9-9b0d2bf26d96" />
